Woof/WoofTests/NetworkLayer/Endpoint/Encodable+AsDictionaryTests.swift

```diff
@@ -0,0 +1,80 @@
+import XCTest
+
+final class EncodableAsDictionaryTests: XCTestCase {
+    func testAsDictionaryMethodSuccessfullyConvertedObjectToDictionary() {
+        // Given
+        struct Model: Encodable {
+            let name: String
+            let age: Int
+        }
+
+        let testModel = Model(name: "John Doe", age: 30)
+
+        do {
+            // When
+            let dictionary = try testModel.asDictionary()
+
+            // Then
+            XCTAssertEqual(dictionary["name"] as? String, "John Doe")
+            XCTAssertEqual(dictionary["age"] as? Int, 30)
+        } catch {
+            XCTFail("Failed to convert the model to a dictionary: \(error)")
+        }
+    }
```

Comment:
I think we don't need `do-catch` because we don't handle errors here. 
```suggestion
    func testAsDictionaryMethodSuccessfullyConvertedObjectToDictionary() throws {
        // Given
        struct Model: Encodable {
            let name: String
            let age: Int
        }

        let testModel = Model(name: "John Doe", age: 30)

            // When
            let dictionary = try testModel.asDictionary()

            // Then
            XCTAssertEqual(dictionary["name"] as? String, "John Doe")
            XCTAssertEqual(dictionary["age"] as? Int, 30)
    }
```

Reply:
Fixed